cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/lineage/field/FieldLineageInfo.java

```diff
@@ -313,55 +317,63 @@ private long computeChecksum() {
     for (WriteOperation write : writeOperations) {
       List<InputField> inputs = write.getInputs();
       for (InputField input : inputs) {
-        Set<String> visitedOperationNames = new LinkedHashSet<>();
-        visitedOperationNames.add(write.getName());
-        computeIncomingSummaryHelper(new EndPointField(write.getDestination(), input.getName()),
-                                     operationsMap.get(input.getOrigin()), write, visitedOperationNames, result);
+        Set<Operation> incomingOperations = new LinkedHashSet<>();
+        incomingOperations.add(write);
+        computeIncoming(new EndPointField(write.getDestination(), input.getName()),
+                        operationsMap.get(input.getOrigin()), write, incomingOperations, result);
       }
     }
     return result;
   }
 
-  private void computeIncomingSummaryHelper(EndPointField destination, Operation currentOperation,
-                                            Operation previousOperation, Set<String> visitedOperationNames,
-                                            Map<EndPointField, Set<EndPointField>> result) {
-    if (!visitedOperationNames.add(currentOperation.getName())) {
+  /**
+   * Helper method to compute the incoming connections
+   *
+   * @param destination the {@link EndPointField} whose incoming connection needs to be processed
+   * @param currentOperation the operation being processed (since we are processing incoming this operation is on the
+   * left side if graph is imagined in horizontal orientation or this operation is the input to the to
+   * previousOperation
+   * @param previousOperation the previous operation which is processed and reside on right to the current operation if
+   * the graph is imagined to be in horizontal orientation.
+   * @param incomingOperations all the operations which are processed so far
+   * @param result a {@link Map} of {@link EndPointField} to {@link Set} of {@link EndPointField} which represents all
+   * the fields which have incoming connection the key field
+   */
+  private void computeIncoming(EndPointField destination, Operation currentOperation,
+                               Operation previousOperation, Set<Operation> incomingOperations,
```

Comment:
I think the previous name was better. 'incomingOperations' sounds like the operations that are input to the current operation.